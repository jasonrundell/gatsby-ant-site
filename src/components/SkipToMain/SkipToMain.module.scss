@import '../../styles/theme';

.link {
  left: -999px;
  position: absolute;
  top: auto;
  width: 1px;
  height: 1px;
  overflow: hidden;
  z-index: -999;
  &:active,
  &:focus {
    color: $white;
    background-color: $dark;
    left: auto;
    top: auto;
    width: 50%;
    height: auto;
    overflow: auto;
    margin: 1rem 25%;
    padding: 0.375rem 0;
    border-radius: 1rem;
    border: 0.375rem solid $dark;
    text-align: center;
    font-size: 1rem;
    z-index: 999;
  }
}
