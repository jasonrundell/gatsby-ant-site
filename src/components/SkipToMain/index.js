import React from 'react'

import styles from './SkipToMain.module.scss'

export default () => (
  <a href="#main" className={styles.link}>
    Skip to main content
  </a>
)
