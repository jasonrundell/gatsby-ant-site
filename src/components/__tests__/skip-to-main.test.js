import React from 'react'
import renderer from 'react-test-renderer'

import SkipToMain from '../skip-to-main'

describe('SkipToMain', () => {
  it('renders correctly', () => {
    const tree = renderer.create(<SkipToMain />).toJSON()
    expect(tree).toMatchSnapshot()
  })
})
