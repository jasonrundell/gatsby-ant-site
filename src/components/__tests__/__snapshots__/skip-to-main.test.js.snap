// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SkipToMain renders correctly 1`] = `
<a
  className="link"
  href="#main"
>
  Skip to main content
</a>
`;
